src/sentry/api/endpoints/organization_member_invite/details.py

```diff
@@ -61,9 +122,99 @@ def get(
         return Response(serialize(invited_member, request.user))
 
     def put(
-        self, request: Request, organization: Organization, invited_member: OrganizationMemberInvite
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
     ) -> Response:
-        raise NotImplementedError
+        """
+        Update an invite request to Organization
+        ````````````````````````````````````````
+
+        Update and/or approve an invite request to an organization.
+
+        :pparam string organization_id_or_slug: the id or slug of the organization the member will belong to
+        :param string member_id: the member ID
+        :param boolean approve: allows the member to be invited
+        :param string orgRole: the suggested org-role of the new member
+        :param array teams: the teams which the member should belong to.
+        :auth: required
+        """
+        if not features.has(
+            "organizations:new-organization-member-invite", organization, actor=request.user
+        ):
+            return Response({"detail": MISSING_FEATURE_MESSAGE}, status=403)
+        # if the requesting user doesn't have >= member:admin perms, then they cannot approve invite
+        # members can reinvite users they invited, but they cannot edit invite requests
+        allowed_roles = get_allowed_org_roles(request, organization)
+        validator = OrganizationMemberInviteRequestValidator(
+            data=request.data,
+            partial=True,
+            context={
+                "organization": organization,
+                "allowed_roles": allowed_roles,
+                "org_role": invited_member.role,
+                "teams": invited_member.organization_member_team_data,
+            },
+        )
+        if not validator.is_valid():
+            return Response(validator.errors, status=400)
+
+        result = validator.validated_data
+
+        is_member = not request.access.has_scope("member:admin") and (
+            request.access.has_scope("member:invite")
+        )
+        members_can_invite = not organization.flags.disable_member_invite
+        # Members can only resend invites
+        is_reinvite_request_only = (
+            set(result.keys()).issubset({"reinvite", "regenerate"})
+            and "approve" not in request.data
+        )
+
+        # Members can only resend invites that they sent
+        is_invite_from_user = invited_member.inviter_id == request.user.id
+
+        if is_member:
+            if not (members_can_invite and is_reinvite_request_only):
+                # this check blocks members from doing anything but reinviting
+                raise PermissionDenied
+            if not is_invite_from_user:
+                return Response({"detail": ERR_MEMBER_INVITE}, status=403)
```

Comment:
should we be checking member permissions before validating the incoming data? i would also put this into its own function if possible